File: backend/app/core/phone.py
import re
from typing import Optional


def normalize_phone(phone: Optional[str]) -> Optional[str]:
    if not phone:
        return None
    cleaned = re.sub(r"[^\d+]", "", phone.strip())
    if not cleaned:
        return None
    if cleaned.startswith("00"):
        cleaned = f"+{cleaned[2:]}"
    if cleaned.startswith("+"):
        digits = re.sub(r"\D", "", cleaned)
        return f"+{digits}"
    digits = re.sub(r"\D", "", cleaned)
    if digits.startswith("41"):
        return f"+{digits}"
    if digits.startswith("0") and len(digits) == 10:
        return f"+41{digits[1:]}"
    return None


def is_valid_swiss_phone(value: Optional[str]) -> bool:
    if not value:
        return False
    digits = re.sub(r"\D", "", value)
    return digits.startswith("41") and len(digits) == 11

File: frontend/lib/phone.ts
export const normalizePhone = (value: string) => {
    const cleaned = value.replace(/\s+/g, '')
    if (!cleaned) return ''
    if (cleaned.startsWith('+')) return cleaned
    if (cleaned.startsWith('00')) return `+${cleaned.slice(2)}`
    if (cleaned.startsWith('0')) return `+41${cleaned.slice(1)}`
    if (cleaned.startsWith('41')) return `+${cleaned}`
    return cleaned
}

export const isValidSwissPhone = (value: string) => {
    if (!value) return true
    if (!value.startsWith('+41')) return false
    const digits = value.replace(/\D/g, '')
    return digits.length === 11
}

export const formatSwissPhone = (value: string) => {
    const digits = value.replace(/\D/g, '')
    if (!digits) return ''
    let rest = ''
    if (digits.startsWith('41')) {
        rest = digits.slice(2)
    } else if (digits.startsWith('0')) {
        rest = digits.slice(1)
    } else {
        rest = digits
    }
    rest = rest.slice(0, 9)
    const seg1 = rest.slice(0, 2)
    const seg2 = rest.slice(2, 5)
    const seg3 = rest.slice(5, 7)
    const seg4 = rest.slice(7, 9)
    let formatted = '+41'
    if (seg1) formatted += ` ${seg1}`
    if (seg2) formatted += ` ${seg2}`
    if (seg3) formatted += ` ${seg3}`
    if (seg4) formatted += ` ${seg4}`
    return formatted
}
